### PR TITLE
Start wrapping errors and panicing less

### PR DESCRIPTION
**Pull request by sectioneight**
GFM-156

### CODE REVIEW

**Comment by sectioneight** [comment]
Tip builds are currently broken due to a change in the context library. 1.7 is passing though.

@uber-go/service-framework

**Comment by sectioneight** [comment]
Fix context issue in #48

**Comment by sectioneight** [comment]
Build is fixed.

**Comment by sectioneight** [comment]
cc @shawnburke

File: core/config/value.go [hunk 1]
```diff
@@ -443,7 +444,7 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, err
 			if v2 := global.GetValue(childKey); v2.HasValue() {
 
 				if err := v2.PopulateStruct(newTarget.Interface()); err != nil {
-					return nil, err
+					return nil, errors.Wrap(err, "unable to populate struct of object target")
```
Comment (madhuravi) [inline core/config/value.go]: Why are error messages not capitalized?

Comment (sectioneight) [inline core/config/value.go]: errors are never capitalized. they're meant to he interpolated into log messages

**Pull request merged**